hypershift-operator/controllers/hostedcluster/internal/platform/powervs/ccm.go

```diff
@@ -0,0 +1,294 @@
+package powervs
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/IBM/platform-services-go-sdk/iamidentityv1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+	"text/template"
+
+	ibmcore "github.com/IBM/go-sdk-core/v5/core"
+)
+
+const (
+	// ConfigMap
+	configMapName         = "cloud-conf"
+	kubeConfigKeyInSecret = "kubeconfig"
+
+	// Deployment
+	deploymentName         = "ibm-cloud-controller-manager"
+	containerName          = "cloud-controller-manager"
+	kubeConfigFileBasePath = "/etc/kubernetes"
+	secretMountPath        = "/etc/vpc"
+	configMapMountPath     = "/etc/ibm"
+)
+
+type ConfigMap struct {
+	Version                  string
+	ConfigFile               string
+	ClusterDefaultProvider   string
+	AccountID                string
+	ClusterID                string
+	G2workerServiceAccountID string
+	G2Credentials            string
+	G2ResourceGroupName      string
+	G2VpcSubnetNames         string
+	G2VpcName                string
+	Region                   string
+	PowerVSCloudInstanceID   string
+	PowerVSRegion            string
+	PowerVSZone              string
+}
+
+func (ccm setupCCM) getAccountID() (accountID string, err error) {
+	var secret corev1.Secret
+	err = ccm.c.Get(ccm.ctx, client.ObjectKey{Namespace: ccm.hcluster.GetNamespace(), Name: ccm.hcluster.Spec.Platform.PowerVS.ClusterControlManagerCreds.Name}, &secret)
+	if err != nil {
+		err = fmt.Errorf("failed to get control plane operator provider creds %s: %w", ccm.hcluster.Spec.Platform.PowerVS.ClusterControlManagerCreds.Name, err)
+		return
+	}
+	secretData, secretHasData := secret.Data["ibmcloud_api_key"]
+	if !secretHasData {
+		err = fmt.Errorf("cluster control manager credentials secret %q is missing credentials key", secret.Name)
+		return
+	}
+	apiKeyStr := string(secretData)
+	auth := &ibmcore.IamAuthenticator{
+		ApiKey: apiKeyStr,
+	}
+
+	iamv1, err := iamidentityv1.NewIamIdentityV1(&iamidentityv1.IamIdentityV1Options{
+		Authenticator: auth,
+	})
+	if err != nil {
+		return
+	}
+
+	apiKeyDetailsOpt := iamidentityv1.GetAPIKeysDetailsOptions{IamAPIKey: &apiKeyStr}
+	apiKey, _, err := iamv1.GetAPIKeysDetails(&apiKeyDetailsOpt)
+	if err != nil {
+		return
+	}
+	if apiKey == nil {
+		err = fmt.Errorf("could not retrieve account id")
+		return
+	}
+
+	accountID = *apiKey.AccountID
+	return
+}
+
+func (ccm setupCCM) setupConfigMap() (err error) {
+	accountID, err := ccm.getAccountID()
+	if err != nil {
+		err = fmt.Errorf("error retrieving accountID while setting up config map: %w", err)
+		return
+	}
+
+	configMap := ConfigMap{
+		Version:                  "1.1.0",
```

Comment:
if this not feed by the user then it is good to hardcode in the template itself

Reply:
hardcoded.